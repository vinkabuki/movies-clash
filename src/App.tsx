import React, { useEffect, useState } from 'react'

import axios from 'axios'
import Item from './Item'

function App() {
  const [count, setCount] = useState(0)
  const [movies, setMovies] = useState([])
  const [currentItem, setCurrentItem] = useState(null)

  useEffect(() => {
    axios.get('http://localhost:8000/recommendations/').then((response) => {
      setMovies(response.data)
      setCurrentItem(response.data[0])
    })
  }, [])

  useEffect(() => {
    if (currentItem !== null) {
      setCurrentItem(movies[count])
    }
  }, [currentItem, movies, count])

  const onAccept = (id) => {
    axios.put(`http://localhost:8000/accept/`, { id: id })
    setCount(count + 1)
  }

  const onReject = (id) => {
    axios.put(`http://localhost:8000/recommendations/${id}/reject/`)
    setCount(count + 1)
  }

  if (currentItem === null) {
    return <div>wait</div>
  } else if (count === movies.length) {
    return <div>there is no more movies</div>
  } else {
    return (
      <div className="App">
        <Item
          id={currentItem.id}
          imageURL={currentItem.imageURL}
          title={currentItem.title}
          summary={currentItem.summary}
          rating={currentItem.rating}
          onAccept={onAccept}
          onReject={onReject}
        ></Item>
      </div>
    )
  }
}

export default App
