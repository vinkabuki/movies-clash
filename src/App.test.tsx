import React from 'react'
import ShallowRenderer from 'react-test-renderer/shallow'
import App from './App'

describe('App component basic shallow render test', () => {
  test('Render App component', () => {
    const renderer = new ShallowRenderer()
    renderer.render(<App />)
    const component = renderer.getRenderOutput()
    expect(component).toMatchSnapshot()
  })
})
