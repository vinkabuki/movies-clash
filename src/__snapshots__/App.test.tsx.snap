// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component basic shallow test Render main component 1`] = `
<div>
  waiting for data
</div>
`;
